src/main/java/com/google/step/snippet/data/Card.java

```diff
@@ -28,4 +28,28 @@ public String getLink() {
   public String getDescription() {
     return description;
   }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
Quick note, at the beginning of your equals method, add a self-check so if the objects are the same, you can return true:
```
if (this == obj) {
  return true;
}
```